StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/Helpers/FixAllContextHelper.cs

```diff
@@ -0,0 +1,228 @@
+                        }
+                    });
+
+                    return projectsAndDiagnostics.ToImmutableDictionary();
+                }
+            }
+
+            return ImmutableDictionary<Project, ImmutableArray<Diagnostic>>.Empty;
+        }
+
+        private static ImmutableDictionary<string, ImmutableArray<Type>> GetAllAnalyzers()
+        {
+            Assembly assembly = typeof(NoCodeFixAttribute).GetTypeInfo().Assembly;
+
+            var diagnosticAnalyzerType = typeof(DiagnosticAnalyzer);
+
+            var analyzers = ImmutableDictionary.CreateBuilder<string, ImmutableArray<Type>>();
+
+            foreach (var type in assembly.DefinedTypes)
+            {
+                if (type.IsSubclassOf(diagnosticAnalyzerType) && !type.IsAbstract)
+                {
+                    Type analyzerType = type.AsType();
+                    DiagnosticAnalyzer analyzer = (DiagnosticAnalyzer)Activator.CreateInstance(analyzerType);
+                    foreach (var descriptor in analyzer.SupportedDiagnostics)
+                    {
+                        ImmutableArray<Type> types;
+                        if (analyzers.TryGetValue(descriptor.Id, out types))
+                        {
+                            types = types.Add(analyzerType);
+                        }
+                        else
+                        {
+                            types = ImmutableArray.Create(analyzerType);
+                        }
+
+                        analyzers[descriptor.Id] = types;
+                    }
+                }
+            }
+
+            return analyzers.ToImmutable();
+        }
+
+        private static ImmutableArray<DiagnosticAnalyzer> GetDiagnosticAnalyzersForContext(FixAllContext fixAllContext)
+        {
+            return DiagnosticAnalyzers
+                .Where(x => fixAllContext.DiagnosticIds.Contains(x.Key))
+                .SelectMany(x => x.Value)
+                .Distinct()
+                .Select(type => (DiagnosticAnalyzer)Activator.CreateInstance(type))
+                .ToImmutableArray();
+        }
+
+        private static async Task<ImmutableArray<Diagnostic>> GetAllDiagnosticsAsync(FixAllContext fixAllContext, Project project, ImmutableArray<DiagnosticAnalyzer> analyzers)
+        {
+            bool includeCompilerDiagnostics = fixAllContext.DiagnosticIds.Any(x => x.StartsWith("CS", StringComparison.Ordinal));
+            var compilation = await project.GetCompilationAsync(fixAllContext.CancellationToken).ConfigureAwait(false);
+
+            var diagnostics = ImmutableArray<Diagnostic>.Empty;
+            if (analyzers.Any())
+            {
+                var compilationWithAnalyzers = compilation.WithAnalyzers(analyzers, project.AnalyzerOptions, fixAllContext.CancellationToken);
+                diagnostics = await compilationWithAnalyzers.GetAnalyzerDiagnosticsAsync().ConfigureAwait(false);
```

Comment:
@mavasani I'm finding that when I use `CompilationWithAnalyzers` here along with #1979, even though all diagnostics are reported in the errors pane in Visual Studio not all diagnostics end up corrected by the Fix All in Project and Fix All in Solution operations. Is there some way I can update this method so we have the performance of `CompilationWithAnalyzers.GetAnalyzerDiagnosticsAsync`, but also get all diagnostics?